BUG: `stats`: regression in JAX 0.6.2

### Describe your issue.

A test started failing after upgrading from jax 0.6.1 to 0.6.2.

Note that scipy CPU CI uses pypi, which automatically pulls in 0.6.2, whereas both scipy GPU CI and https://github.com/rgommers/pixi-dev-scipystack use conda, which is still at 0.6.0 even after you run `pixi update jax jaxlib`.

### Reproducing Code Example

```python
To reproduce locally with https://github.com/rgommers/pixi-dev-scipystack:


diff --git a/scipy/pixi.toml b/scipy/pixi.toml
index 0e6e5a8..0e4252d 100644
--- a/scipy/pixi.toml
+++ b/scipy/pixi.toml
@@ -173,9 +173,9 @@ test-cupy = { cmd = "spin test -b cupy -m 'array_api_backends and not slow'", cw
 # Windows support pending: https://github.com/conda-forge/jaxlib-feedstock/issues/161
 platforms = ["linux-64", "osx-arm64"]
 
-[feature.jax-cpu.dependencies]
-jax = ">=0.6.0"
-jaxlib = { version = "*", build = "*cpu*" }
+[feature.jax-cpu.pypi-dependencies]
+jax = "==0.6.2"
+jaxlib = "==0.6.2"
 
 [feature.jax-cuda]
 platforms = ["linux-64"]


pixi r test-jax -t scipy.stats -- -k test_ddof_equal_to_number_of_observations

The above passes with 0.6.0 (conda) and 0.6.1 (pixi), but fails with 0.6.2.
```

### Error message

```shell
e.g. https://github.com/scipy/scipy/actions/runs/15727179767/job/44319546946?pr=23175


________ TestGSTD.test_ddof_equal_to_number_of_observations[jax.numpy] _________

self = <scipy.stats.tests.test_stats.TestGSTD object at 0x7f93248e73e0>
xp = <module 'jax.numpy' from '/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/jax/numpy/__init__.py'>

    def test_ddof_equal_to_number_of_observations(self, xp):
        x = xp.asarray(self.array_1d)
        res = stats.gstd(x, ddof=x.shape[0])
>       xp_assert_equal(res, xp.asarray(xp.nan))
E       AssertionError: 
E       Arrays are not equal
E       
E       nan location mismatch:
E        ACTUAL: array(0.)
E        DESIRED: array(nan)

res        = Array(0., dtype=float64)
self       = <scipy.stats.tests.test_stats.TestGSTD object at 0x7f93248e73e0>
x          = Array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,
       18, 19, 20, 21, 22, 23, 24], dtype=int64)
xp         = <module 'jax.numpy' from '/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/jax/numpy/__init__.py'>

scipy/stats/tests/test_stats.py:7053: AssertionError
```

### SciPy/NumPy/Python version and system information

```shell
git tip as of today
```

--- a/scipy/stats/tests/test_stats.py
+++ b/scipy/stats/tests/test_stats.py
@@ -7060,7 +7060,6 @@ class TestGSTD:
         gstd_actual = stats.gstd(a, axis=1)
         xp_assert_close(gstd_actual, xp.asarray([4, np.nan]))
 
-    @xfail_xp_backends("jax.numpy", reason="returns subnormal instead of nan")
     def test_ddof_equal_to_number_of_observations(self, xp):
         x = xp.asarray(self.array_1d)
         res = stats.gstd(x, ddof=x.shape[0])

